helper/storagepacker/storagepacker.go

```diff
@@ -125,112 +125,82 @@ func (s *StoragePacker) BucketKey(itemID string) string {
 
 // DeleteItem removes the item from the respective bucket
 func (s *StoragePacker) DeleteItem(_ context.Context, itemID string) error {
-	return s.DeleteMultipleItems(context.Background(), nil, itemID)
+	return s.DeleteMultipleItems(context.Background(), nil, []string{itemID})
 }
 
-func (s *StoragePacker) DeleteMultipleItems(ctx context.Context, logger hclog.Logger, itemIDs ...string) error {
-	var err error
-	switch len(itemIDs) {
-	case 0:
-		// Nothing
+func (s *StoragePacker) DeleteMultipleItems(ctx context.Context, logger hclog.Logger, itemIDs []string) error {
+	if len(itemIDs) == 0 {
 		return nil
+	}
 
-	case 1:
+	if logger == nil {
 		logger = hclog.NewNullLogger()
-		fallthrough
-
-	default:
-		lockIndexes := make(map[string]struct{}, len(s.storageLocks))
-		for _, itemID := range itemIDs {
-			bucketKey := s.BucketKey(itemID)
-			if _, ok := lockIndexes[bucketKey]; !ok {
-				lockIndexes[bucketKey] = struct{}{}
-			}
-		}
+	}
 
-		lockKeys := make([]string, 0, len(lockIndexes))
-		for k := range lockIndexes {
-			lockKeys = append(lockKeys, k)
+	// Sort the ids by the bucket the will be deleted from
+	lockKeys := make([]string, 0)
+	byBucket := make(map[string]map[string]struct{})
+	for _, id := range itemIDs {
+		bucketKey := s.BucketKey(id)
+		bucket, ok := byBucket[bucketKey]
+		if !ok {
+			bucket = make(map[string]struct{})
 		}
 
-		locks := locksutil.LocksForKeys(s.storageLocks, lockKeys)
-		for _, lock := range locks {
-			lock.Lock()
-			defer lock.Unlock()
-		}
+		bucket[id] = struct{}{}
+		byBucket[bucketKey] = bucket
+		lockKeys = append(lockKeys, bucketKey)
```

Comment:
We might be adding more bucket keys to the list here than necessary. The LocksForKeys function doesn't check for duplicates. I think we'll end up with the right number of locks eventually, but with more memory utilization than required. WDYT?

Reply:
Updated, let me know what you think